lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -520,13 +520,13 @@ internal class EvaluatingCompiler(
     /**
      *  For operators which could return integer type, check integer overflow in case of [TypingMode.PERMISSIVE].
      */
-    private fun resolveArithmeticOverflow(computeThunk: ThunkEnv, metas: MetaContainer): ThunkEnv =
+    private fun checkIntegerOverflow(computeThunk: ThunkEnv, metas: MetaContainer): ThunkEnv =
         when (val staticTypes = metas.staticType?.type?.getTypes()) {
             // No staticType, can't validate integer size.
             null -> computeThunk
             else -> {
-                when (compileOptions.typingMode) {
-                    TypingMode.LEGACY -> {
+                when (compileOptions.typedOpBehavior) {
```

Comment:
First of all, I think we want to always check if the result integer value overflows or not. It shouldn't be a configurable option. 

The reason why I want to change `TypingMode` to `TypedOpBehavior` is that integer constraints are modeled as type parameters currently, https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/types/StaticType.kt#L379 (even though I don't think this is a right way to model integer types). So the integer overflow checking semantically lines up better with whether type parameters work (`TypedOpBehavior`), instead of whether we should return `NULL`, `MISSING` or throw error (TypingMode).

Reply:
Could we save this change for a separate discussion and/or PR? I don't want to conflate changing the default `TypedOpBehavior` with a possibly breaking behavioral change.

I generally agree that the `TypingMode` should only really dictate the data type mismatches behaviors for clauses & operators. I think the prior `when` expression branching was only to verify that the overflow checking was done in `PERMISSIVE` mode because the static typing was only done in `PERMISSIVE` mode and not yet in `LEGACY` yet. See comments: https://github.com/partiql/partiql-lang-kotlin/blob/6c9949c0ac26806b20cc5dd42619b81fa29b099a/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L520-L533

That's why I don't think you can do a 1:1 drop-in replacement of `TypingMode.LEGACY` for `TypedOpBehavior.LEGACY`.